docs/editor/variables-reference.md

```diff
@@ -243,6 +243,10 @@ Command inputs can also be used with tasks. In this example, the built-in Termin
 
 ## Common questions
 
+### When does variable substitution occur in configuration parsing?
+
+Only after all variables in the configuration are evaluated, will the variables be substituted with their (string) result.  Duplicate variables are also ignored and will receive the result of the first evaluation.
+
```

Comment:
I suggest to clearly spell out how and why it is done: 

```
### Details of variable substitution in a debug configurations or tasks

Variable substitution in debug configurations or tasks is a 2-pass process:
- in the first pass all variables are evaluated to string results. If a variable occurs more than once, it is only evaluated once.
- in the second pass all variables are substituted with the results from the first pass.

A consequence of this is that the evaluation of a variable (e.g. a command-based variable implemented in an extension) has **no access** to other substituted variables in the debug configuration or task. It only sees the original variables. This means that variables cannot depend on each other (which ensures isolation and makes substitution robust against evaluation order).
```

Reply:
I like this.